### PR TITLE
[cleanup] Move _redirects from public to static

### LINKED ISSUE #43: [redirects] Don't version control files under the auto-generated `public` directory
Currently the Netlify `_redirects` file is being edited directly from the Hugo-generated `public` directory. It should not be setup in that way.

Instead move the file to `static/_redirects`.

Later we can consider switching to using Hugo aliases, and eventually an implementation of Hugo aliases via Netlify redirects as has been done elsewhere, such as https://github.com/open-telemetry/opentelemetry.io/issues/781.

Waiting for the following to be merged first:

- [x] #23
- [x] #38

### PR DESCRIPTION
**Pull request by chalin**
- Fixes #43
  - Moves `_redirects` from `public` to `static`
- Adjusts `layouts/partials/css.html` so that it creates a production build only in production (when `hugo.IsProduction` is true)
- Adds NPM scripts to build and serve the website. Note that these scripts aren't used by the Netlify build commands yet.

### Redirect tests

- https://deploy-preview-51--in-toto.netlify.app/Statement/v0.1.0
- https://deploy-preview-51--in-toto.netlify.app/attestation/vulns/v...

### CODE REVIEW

**Comment by chalin** [comment]
@DarikshaAnsari - we'll need to ensure that the `_redirects` file is also present in the Docsy-based version of the website.

**Review by lukpueh — COMMENTED** [review]
>   * Moves `_redirects` from `public` to `static`
👍 

> * Adjusts `layouts/partials/css.html` so that it creates a production build only in production (when `hugo.IsProduction` is true)

TBH, I can't tell if this is correct. Should I see a difference in the preview build? It looks like it does use the production css link (w/ hash):
```
<link rel="stylesheet" href="/css/style.94f9acd6bd00126cfa5eeb849dbf3cc631715b317a956d7516a0d3e4e1c1cef7.css" integrity="sha256-lPms1r0AEmz6XuuEnb88xjFxWzF6lW11FqDT5OHBzvc=">
```

> * Adds NPM scripts to build and serve the website. Note that these scripts aren't used by the Netlify build commands yet.

Are these commands also for devs to build locally? If so, should we update our CONTRIBUTING.md? It does not have to be part of this pr.

> * https://deploy-preview-51--in-toto.netlify.app/Statement/v0.1.0
> * https://deploy-preview-51--in-toto.netlify.app/attestation/vulns/v...

💯 for providing links here.

**Comment by chalin** [comment]
> > * Adjusts `layouts/partials/css.html` so that it creates a production build only in production (when `hugo.IsProduction` is true)
> 
> TBH, I can't tell if this is correct. Should I see a difference in the preview build? It looks like it does use the production css link (w/ hash):

Right, I had not enabled the scripts for use by Netlify. I've added that change so that you can see the effect by viewing source of the preview. Let me know what you think.

> > * Adds NPM scripts to build and serve the website. Note that these scripts aren't used by the Netlify build commands yet.
> 
> Are these commands also for devs to build locally? If so, should we update our CONTRIBUTING.md? It does not have to be part of this pr.

Both for use by Netlify and devs. I've updated `CONTRIBUTING.md` too.

> > * https://deploy-preview-51--in-toto.netlify.app/Statement/v0.1.0
> > * https://deploy-preview-51--in-toto.netlify.app/attestation/vulns/v...
> 
> 💯 for providing links here.

👍🏻

File: package.json [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+  "scripts": {
+    "_build": "npm run _hugo-dev --",
+    "_hugo": "hugo --cleanDestinationDir",
+    "_hugo-dev": "npm run _hugo -- -e dev -DF",
+    "_serve": "npm run _hugo-dev -- --minify --disableFastRender serve",
+    "build:preview": "npm run _hugo-dev -- --minify --baseURL \"${DEPLOY_PRIME_URL:-http://localhost}\"",
+    "build:production": "npm run _hugo -- --minify",
+    "build": "npm run _build --",
+    "clean": "rm -Rf node_modules public/* resources",
+    "make:public": "mkdir -p public && git init -b main public",
```
Comment (lukpueh) [inline package.json:11]: What's this command needed for?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -35,17 +35,17 @@ npm install
...
 
-Once Hugo is installed, you can serve the project locally to see your changes in real-time.
-The root of the directory contains a `Makefile` for automating development processes. The `make` CLI tool is usually installed by default on most systems (excluding Windows), but you can check if it is installed by running `make --version` on your terminal. If this command is unsuccessful, you will need to find the standard method for installing it for your system. For installing `make` on Windows, please see [here](https://gnuwin32.sourceforge.net/packages/make.htm).
-Run the following command:
+Once Hugo is installed, you can locally serve the website to see your changes in
+real-time:
 
 ```sh
-make serve
+npm run serve
```
Comment (lukpueh) [inline CONTRIBUTING.md:44]: Did you try this? It fails on my machine with this error:

```
> hugo --cleanDestinationDir -e dev -DF --minify --disableFastRender serve

Error: unknown flag: --disableFastRender
```

Comparing this to the removed Makefile, it looks like it's missing the `server` subcommand.

**Review by lukpueh — CHANGES_REQUESTED** [review]

**Comment by lukpueh** [comment]
> > TBH, I can't tell if this is correct. Should I see a difference in the preview build? It looks like it does use the production css link (w/ hash):
> 
> Right, I had not enabled the scripts for use by Netlify. I've added that change so that you can see the effect by viewing source of the preview. Let me know what you think.

Looks like expected now. Thanks!

File: package.json (hunk 1 above)
Comment (chalin) [reply package.json:11]: It creates the `public` folder and sets it up as a local git repo. This makes it easier to view generated file diff when working on website changes. It's more of a power-user feature. Would you like for me to remove it?

File: CONTRIBUTING.md (hunk 2 above)
Comment (chalin) [reply CONTRIBUTING.md:44]: Fixed. No, it was because of the wrong order of arguments.

File: package.json (hunk 1 above)
Comment (lukpueh) [reply package.json:11]: No, it's fine.

**Review by lukpueh — APPROVED** [review]

**Pull request merged**